multiqc/modules/htstream/apps/AdapterTrimmer.py

```diff
@@ -0,0 +1,167 @@
+from collections import OrderedDict
+import logging
+
+from multiqc import config
+from multiqc.plots import table, bargraph
+
+#################################################
+
+""" AdapterTrimmer submodule for HTStream charts and graphs """
+
+#################################################
+
+class AdapterTrimmer():
+
+	########################
+	# Info about App
+	def __init__(self):
+		self.info = "Trims adapters which are sequenced when the fragment insert length is shorter than the read length."
+		self.type = "bp_reducer"
+
+
+	########################
+	# Table Function
+	def table(self, json, total, zeroes, index):
+
+		# Table constructor. Just like the MultiQC docs.
+
+		# If total is zero, no need for a plot
+		if total == 0:
+			return ""
+		
+		headers = OrderedDict()
+
+		# Some columns have SUPER small values, add raw counts instead of percentages
+		if zeroes == False:
+			headers["At_%_BP_Lost" + index] = {'title': "% Bp Lost",
+										'namespace': "% Bp Lost",
+										'description': 'Percentage of Input bps (SE and PE) trimmed.',
+										'suffix': '%',
+										'format': '{:,.2f}',
+										'scale': 'RdPu'
+										}
+			headers["At_%_Adapters" + index] = {'title': "% Adapters",
+										'namespace': "% Adapters",
+										'description': 'Percentage of Reads (SE and PE) with an Adapter',
+										'suffix': '%',
+										'format': '{:,.2f}',
+										'scale': 'Blues'
+										}
+		
+		else:
+			headers["At_BP_Lost" + index] = {'title': "Bp Lost", 'namespace': "Bp Lost", 'description': 'Input bps (SE and PE) trimmed.', 'scale': 'RdPu', 'format': '{:,.0f}'}
+			headers["At_Adapters" + index] = {'title': "Adapters", 'namespace': "Adapters", 'description': 'Reads (SE and PE) with an Adapter', 'scale': 'Blues', 'format': '{:,.0f}'}
+
+		# More columns
+		headers["At_Avg_BP_Trimmed" + index] = {'title': "Avg. Bps Trimmed", 'namespace': "Avg. Bps Trimmed", 'description': 'Average Number of basepairs trimmed from reads', 'format': '{:,.2f}', 'scale': 'Oranges'}
+		headers["At_Notes" + index] = {'title': "Notes", 'namespace': "Notes", 'description': 'Notes'}
+
+		return table.plot(json, headers)
+
+
+	########################
+	# Bargraphs Function
+	def bargraph(self, json, avg_bp_trimmed):
+
+		# configuration dictionary for bar graph
+		config = {'title': "HTStream: Trimmed Bp Composition Bargraph",
+				  'id': "htstream_adaptertrimmer_bargraph",
+				  'ylab' : "Basepairs",
+				  'cpswitch_c_active': True}
+
+		# Title
+		html = "<h4> AdapterTrimmer: Trimmed Basepairs Composition </h4>\n" 
+		html += "<p>Composition of basepairs trimmed from the ends of paired end and single end reads.</p>"
+
+
+		# if no overlaps at all are present, return nothing
+		if avg_bp_trimmed == 0:
+			html += '<div class="alert alert-info"> <strong>Notice:</strong> No adapters were trimmed from samples. </div>'	
+			return html
+
+		# If too many samples, forget about it.
+		if len(json.keys()) > 150:
```

Comment:
Any specific reason for this?

Generally your module shouldn't make decisions like this - firstly, MultiQC has a bunch of mechanisms in place to make large bargraphs usable. Second, a lot of people use the data export files generated from MultiQC for downstream analysis even if the plots in the report are unreadable. These files are not generated if you exit early like this.